### PR TITLE
Some initial analytics for PAL usage

### PR DESCRIPTION
**Pull request by davidkelly**
I track a few things:  
* duration the PAL was open
* select of an avatar (with the target uuid)
* ban, silence, mute, ignore (with the target uuid)
* refresh of pal
* display name change
* volume slider change 

Test Plan
Well first, you need to have a metabase account.  If you do, then open the pal, click avatars and buttons to your heart's content.  Then, close the PAL.  Then do a query on the User Activities table, restricting to your user id.  Wait a really long time, and you will see the results.  The pal_action and pal_opened stats should all be in there.

### CODE REVIEW

File: libraries/networking/src/UserActivityLoggerScriptingInterface.cpp [hunk 1]
```diff
@@ -38,6 +38,21 @@ void UserActivityLoggerScriptingInterface::tutorialProgress( QString stepName, i
...
+void UserActivityLoggerScriptingInterface::palAction(QString action, QString target) {
+    QJsonObject payload;
+    payload["action"] = action;
+    if (target.length() > 0) {
+        payload["target"] = target;
+    }
+    logAction("pal_activity", payload);
+}
+
+void UserActivityLoggerScriptingInterface::palOpened(float secondsOpened) {
```
Comment (zfox23) [inline libraries/networking/src/UserActivityLoggerScriptingInterface.cpp]: If a user opens the PAL then quits Interface, will this log ever get recorded?

**Comment by hifi-gustavo** [comment]
Build console output for commit: d9fbf832e248bb96fc06a6b22c62bf3f93d1468e available at: https://gist.github.com/fb3077620b445075ef7dd1955e207e95<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9441-d9fbf83.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9441-d9fbf83.dmg

File: scripts/system/pal.js [hunk 2]
```diff
@@ -232,14 +233,17 @@ pal.fromQml.connect(function (message) { // messages are {method, params}, like
     case 'refresh':
         removeOverlays();
         populateUserList();
+        UserActivityLogger.palAction("refresh");
         break;
     case 'updateGain':
         data = message.params;
         Users.setAvatarGain(data['sessionId'], data['gain']);
+        UserActivityLogger.palAction("avatar_gain_changed", data['sessionId']);
```
Comment (zfox23) [inline scripts/system/pal.js]: Are we OK with this log getting sent while the user is moving the slider? Do you think it'd be better to only send this log when the user has released their mouse button?

**Comment by zfox23** [comment]
Pretty cool how straightforward this is to implement! I have a couple questions, otherwise LGTM.

File: libraries/networking/src/UserActivityLoggerScriptingInterface.cpp (hunk 1 above)
Comment (davidkelly) [inline libraries/networking/src/UserActivityLoggerScriptingInterface.cpp]: I record it in pal.js - when the script ends (or the dialog box is closed)

File: scripts/system/pal.js (hunk 2 above)
Comment (davidkelly) [inline scripts/system/pal.js]: good point - probably not useful to record it more than once

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5adc2cc593c61b3b01baf80ab13abe2c346072ed available at: https://gist.github.com/fb3077620b445075ef7dd1955e207e95<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9441-5adc2cc.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9441-5adc2cc.dmg

**Pull request merged**